packages/fx-core/src/plugins/resource/identity/utils/telemetryUtil.ts

```diff
@@ -32,24 +29,31 @@ export class TelemetryUtils {
     errorCode: string,
     errorType: string,
     errorMessage: string,
-    properties?: { [key: string]: string },
-    measurements?: { [key: string]: number }
+    properties?: { [key: string]: string; },
+    measurements?: { [key: string]: number; }
   ) {
     if (!properties) {
       properties = {};
     }
-    properties[Telemetry.properties.success] = Telemetry.resultNo;
+    properties[Telemetry.properties.success] = Telemetry.valueNo;
     properties[Telemetry.properties.errorCode] = errorCode;
     properties[Telemetry.properties.errorType] = errorType;
     properties[Telemetry.properties.errorMessage] = errorMessage;
-    properties[Telemetry.properties.component] = Telemetry.componentName;
-    if (this.ctx.app.id) {
-      properties[Telemetry.properties.appid] = this.ctx.app.id;
-    }
+    this.addProperties(properties);
     TelemetryUtils.ctx.telemetryReporter?.sendTelemetryErrorEvent(
       eventName,
       properties,
       measurements
     );
   }
+
+  private static addProperties(properties: { [key: string]: string; }) {
+    properties[Telemetry.properties.component] = Telemetry.componentName;
+    const appId = this.ctx.configOfOtherPlugins.get(Constants.solution)?.get(Constants.remoteTeamsAppId) as string;
+    if (appId) {
```

Comment:
Do we need the following lines here? in 52 it already as string.... So we will provide empty string instead of undefined?

Reply:
We found that it will not raise error here when undefined as string, but we can not find any official doc on it. So we decide to add logic to know whether the value is undefined and assign empty string when undefined, which will have the same result in query.

Reply:
If that is the case line 52 don't need as string right?

Reply:
Yeah, from my side I think we don't need as string in line 52

Reply:
It is fixed, thanks.